package/homebrew.spec

```diff
@@ -0,0 +1,72 @@
+%define homebrew_user linuxbrew
+%define homebrew_directory /home/%{homebrew_user}/.%{homebrew_user}
+%define brew_repo https://github.com/Homebrew/brew
+
+Summary:   Homebrew package manager
+Name:      homebrew
+Version:   0.0.0
+Release:   %autorelease
+License:   BSD-2-clause
+Group:     Development/Tools
+URL:       https://brew.sh
+Source:    %{brew_repo}/archive/refs/tags/%{git_version}.tar.gz#/brew.tar.gz
+
+# See: https://github.com/Homebrew/install/blob/master/install.sh#L211-L214
+BuildRequires: git
+
+Requires: procps-ng
+Requires: file
+Requires: gcc
+Requires: git >= 2.7.0
+Requires: glibc >= 2.13
+Requires: ruby >= 2.6.0
+
+%description
+The Missing Package Manager for macOS (or Linux)
+
+%global debug_package %{nil}
+%global _missing_build_ids_terminate_build 0
+%global __brp_mangle_shebangs /usr/bin/true
+
+%prep
+%autosetup -n brew-%{version}
+
+%build
+git init
+git remote set-url origin %{brew_repo} || git remote add origin %{brew_repo}
+
+%install
+install -d "$RPM_BUILD_ROOT%{homebrew_directory}"
+cp -r . "$RPM_BUILD_ROOT%{homebrew_directory}"
+
+pushd "$RPM_BUILD_ROOT%{homebrew_directory}"
+mkdir -vp Cellar Frameworks etc include lib opt sbin share var/homebrew/linked
+
+%check
```

Comment:
`copr` runs automated checks, and among them is:
```console
+ /usr/lib/rpm/check-buildroot
+ /usr/lib/rpm/redhat/brp-ldconfig
+ /usr/lib/rpm/brp-compress
+ /usr/lib/rpm/brp-strip /usr/bin/strip
+ /usr/lib/rpm/brp-strip-comment-note /usr/bin/strip /usr/bin/objdump
+ /usr/lib/rpm/redhat/brp-strip-lto /usr/bin/strip
+ /usr/lib/rpm/brp-strip-static-archive /usr/bin/strip
+ /usr/lib/rpm/check-rpaths
*******************************************************************************
*
* WARNING: 'check-rpaths' detected a broken RPATH OR RUNPATH and will cause
*          'rpmbuild' to fail. To ignore these errors, you can set the
*          '$QA_RPATHS' environment variable which is a bitmask allowing the
*          values below. The current value of QA_RPATHS is 0x0000.
*
*    0x0001 ... standard RPATHs (e.g. /usr/lib); such RPATHs are a minor
*               issue but are introducing redundant searchpaths without
*               providing a benefit. They can also cause errors in multilib
*               environments.
*    0x0002 ... invalid RPATHs; these are RPATHs which are neither absolute
*               nor relative filenames and can therefore be a SECURITY risk
*    0x0004 ... insecure RPATHs; these are relative RPATHs which are a
*               SECURITY risk
*    0x0008 ... the special '$ORIGIN' RPATHs are appearing after other
*               RPATHs; this is just a minor issue but usually unwanted
*    0x0010 ... the RPATH is empty; there is no reason for such RPATHs
*               and they cause unneeded work while loading libraries
*    0x0020 ... an RPATH references '..' of an absolute path; this will break
*               the functionality when the path before '..' is a symlink
*          
*
* Examples:
* - to ignore standard and empty RPATHs, execute 'rpmbuild' like
*   $ QA_RPATHS=$(( 0x0001|0x0010 )) rpmbuild my-package.src.rpm
* - to check existing files, set $RPM_BUILD_ROOT and execute check-rpaths like
*   $ RPM_BUILD_ROOT=<top-dir> /usr/lib/rpm/check-rpaths
*  
*******************************************************************************
ERROR   0002: file '/home/linuxbrew/.linuxbrew/Library/Homebrew/test/support/fixtures/elf/libforty.so.0' contains an invalid rpath
runpath 'rpath
runpath' in [rpath
runpath]
ERROR   0002: file '/home/linuxbrew/.linuxbrew/Library/Homebrew/test/support/fixtures/elf/hello_with_rpath' contains an invalid rpath '@@HOMEBREW_PREFIX@@/lib' in [@@HOMEBREW_PREFIX@@/lib]
ERROR   0002: file '/home/linuxbrew/.linuxbrew/Library/Homebrew/test/support/fixtures/elf/c.elf' contains an invalid rpath '@@HOMEBREW_PREFIX@@/lib' in [@@HOMEBREW_PREFIX@@/lib]
error: Bad exit status from /var/tmp/rpm-tmp.AebVcP (%install)
    Bad exit status from /var/tmp/rpm-tmp.AebVcP (%install)
```

It seems to complain that the `rpath` are not set to the system install, and I don't think that there is a way to control that check's execution. Will need to ask downstream for suggestions on it. But before that, can someone explain the purpose of those files and why are they pre-compiled and distributed in the git repo?

Reply:
They are files used by `brew tests`

Reply:
Ok, temporarily skipping this test along with the shebang one, so that we can at least run the other checks. I will pick this up again from time to time, but if someone can help in between, feel free to push/force-push to my fork

Reply:
No worries, thanks.